muxers/mplex/src/lib.rs

```diff
@@ -236,3 +205,9 @@ impl AsyncWrite for Substream {
         this.muxer.lock().poll_close_stream(cx, this.id)
     }
 }
+
+impl Drop for Substream {
```

Comment:
Is this change necessary for the general goal of _redesigning the `StreamMuxer` trait_? I would prefer unrelated changes to be split into separate pull requests, thus not having one very large one.

Reply:
I think it is coupled to it, yes. Where-else would you clean up the resources associated with a substream within the mplex implementation?